cli/src/cast.rs

```diff
@@ -80,6 +80,18 @@ async fn main() -> eyre::Result<()> {
             };
             println!("0x{}", output);
         }
+        Subcommands::FromBin { value } => {
+            let val = unwrap_or_stdin(value)?;
+            let hex_string = val
+                .as_bytes()
+                .chunks(4)
+                .map(str::from_utf8)
+                .map(|chunk| u32::from_str_radix(&chunk.unwrap(), 2))
+                .map(|number| format!("{:x}", number.unwrap()))
+                .collect::<Vec<String>>()
+                .join("");
+            println!("0x{}", hex_string)
```

Comment:
This is simpler:

```suggestion
            let hex_string: String = (0..val.len())
                .step_by(8)
                .map(|i| u8::from_str_radix(&val[i..i + 8], 2).expect("Invalid binary"))
                .map(|n| format!("{:x}", n))
                .collect();
            println!("0x{}", hex_string)
```

Unfortunately no way to use `std::fmt::LowerHex` on a `Vec<u8>`, that would have been the cleanest ultimately

Reply:
Although I guess we should actually step by 4 (or 2?) just to be sure

Reply:
@onbjerg if the hex is encoded with 4 bits, that doesn't mean that we **have** to step by 4?

Reply:
No, if we pass 8 bits you just get 2 characters. The amount we can step by is just determined by what kind of input we expect

Reply:
Right, the algo will always convert 4 bits to a char. Thanks!